src/frontend/src/handler/privilege.rs

```diff
@@ -0,0 +1,135 @@
+// Copyright 2022 Singularity Data
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use risingwave_common::ensure;
+use risingwave_common::error::ErrorCode::PermissionDenied;
+use risingwave_common::error::Result;
+use risingwave_pb::user::grant_privilege::{Action as ProstAction, Object as ProstObject};
+use risingwave_sqlparser::ast::{DropStatement, Statement, TableFactor};
+
+use crate::binder::Binder;
+use crate::session::SessionImpl;
+
+/// Convert query Statement `stmt` to privilege we need.
+/// Return `is_owner`(check whether user in `session` is owner of the object), object and action.
+pub(crate) fn resolve_privilege(
```

Comment:
Resolve might have to return multi objects and actions. For example, insert with query source should also take the source into consideration. Besides, query is not covered in this PR, the relation in query may contains `Subquery`/`Join`, these also need to be resolved with `Select` action.

Reply:
Please use `BoundStatement` as parameter of `resolve_privilege`, so you don't have to resolve relation twice.